papers/Raku_Steering_Council_Code.md

```diff
@@ -291,6 +295,136 @@ initial election, will be assumed to be inactive, and have their status
 changed to "inactive", thus revoking their voting rights until they've
 become active again and accepted by a core team vote.
 
+Raku users' community
+---------------------
+
+Anyone who has signed up to the raku users email list is considered a member of the Raku
+users' community.
+
+Raku philosophy
```

Comment:
This section adds a lot of text but very little value to this document. Longer documents are less likely to be read and understood. If you think this is useful, what about creating a separate document that talks about Raku philosophy and other stuff, as a kind of additional reading?

Reply:
The problem of long versus short documents is not easy and also a matter of taste. In principle everyone prefers short to long - so everyone will agree with you in principle! But when short becomes long can vary significantly, and here you will find yourself within a spectrum of people. For me, the final document including all my proposals is not long, nor is it particularly short. 

There are also factors for including material in the basic Governance document, rather than pushing it off to an addendum. By including it in the main Founding Document (all caps here for emphasis), it shows that the Community considers this material to be important. If for example, the matter of elections, the Council, the Core team are in the main document, but the Raku users' community is not, or the Code of Conduct, or the philosophy, then we demonstrate what we consider important and what we consider of secondary importance.

My personal opinion is that in any coordinated grouping of human beings, the most important thing is how we treat each other on a day to day basis, not how we punish each other in extremis. 

If the desire is to make the document shorter, then why not only have the Code of Conduct and the Philosophy, and then push into an addendum everything about election rules, recall procedures, and so on.

But what is important and what is long and short are areas where reasonable people can legitimately disagree.

Reply:
“foo is a matter of taste” is also something everyone will agree with in principle. The practical reality is that we have to somehow navigate through the infinite spectrum of solutions, trying to judge what will be better for our goals.

The document is already too long, you made it even longer. “It may come in handy, put it in” is a very Raku thing to do, but I suggest to focus on things that bring the most value. If you think that the election is secondary and human values are more important, then for all means do that. But there is a need for a useful, clear document that will provide some sort of guidance, don't turn it into a book.

Reply:
@AlexDaniel  Absolutely agree that there is an infinite spectrum, and we need to judge what better serves our goals. That is what we are doing here.

"The document is already too long, [I] made it longer" - true. My favourite literary form is the sonnet, and particularly those by Shakespeare, though I like some Haiku too. So the current document is too long. But it ain't a sonnet.

I also think that the Documentation of the Raku language is not enough, and there are some 700 documents there, the one on Operators is very long. But the governance document we are trying to put together here is not a tutorial or reference. 

"there is a need for a useful, clear document  ..." yes. "that will provide some sort of guidance" - no (to be blunt). A governance document is not about guidance. 

It is a tool to be used in the management of a community of human beings. We are fractious, ambitious, clever, manipulative, truthful, lying, philanthropic, self-centered, and so on. Some people are more of one dimension than another, but to some extent we all have all these tendencies. 

In order to have order and a reference for when matters get heated, or someone tries to impose their will upon others, it has been found repeatedly in many societies that a set of agreed principles or rules should be agreed to **before** they are needed. The Raku community, or the Raku development team is composed of human beings. There is nothing that marks them as being special as humans, however highly we may value the Raku language. So past experience of humanity since history began would indicate that the Raku  community would be well served by a set of rules or principles.

When I said no to 'guidance', I mean a particular semantic attribute. "Guidance" can imply the ability of an individual to behave counter to the guidance. A set of rules for behaviour implies that bad behaviour has consequences for the badly behaving individual. 

It seems to me that the fundamental question here is not about the length of a governance document, or even about guidance versus rules. Its about the freedom to whatever comes into one's head and to express it, without the need to filter the words for consequences.

So lets look at that.

It should be blindingly obvious that the bad behaviour of individuals in any community **always** has consequences. But in order for a community to be sustainable and a friendly place, the bad behaviour of a few has to be sanctioned for the ability of the many to be fruitful. 

There are brands of social thought that emphasis the freedom of the individual to do whatever they want. 

It is my sincere belief that if freedom is taken to excess, communities are unable to cohere. It is a resource question just like time and memory. Spend time, or spend memory. Allow people to behave badly with no restraint, and the number of people who want to take part in such a community diminishes to negligible. Gently constrain excesses of speech, malignant discussions, toxic remarks, and the number of people who are willing to take part increases. 

Personally, I want to be a part of a supportive friendly society, and I am willing for others to tell me that I have erred if that is the entrance fee to such a society. I think it is part of being a mature individual to know that we make stupid mistakes and say bad things, but to accept that we need to pay the consequences for our actions to take the hit for the benefit of everyone else.

The Raku community by and large - in my observation - is a friendly place. I have learnt that some people have been burnt out and rejected the community. That is truly sad. It is something to avoid. A patient helpful community is something that I think every thoughtful and sensitive person would be willing to sacrifice some personal comfort in order to obtain. 

So yes, some documents are long, some are short, a document should be as long as it needs to be and no longer.

Reply:
> I also think that the Documentation of the Raku language is not enough

That is correct. A language riddled with edge cases needs a very long documentation or a very opinionated guide that clearly says “do this, and don't do that”, or both. We have neither. Yes, it is not enough, we need more docs.

> It seems to me that the fundamental question here is not about the length of a governance document, or even about guidance versus rules. Its about the freedom to whatever comes into one's head and to express it, without the need to filter the words for consequences.

No, the fundamental issue here is that you included a bunch of text about language design philosophy, and when I read the text I see very little connection to this repository. For example:

> In particular, this often means replacing imperative step-by-step
mechanisms with approaches that are either entirely declarative,
purely functional, or cleanly object-oriented. And ensuring that these
better approaches were also safe to use together and simple to compose.

Yeah, alright alright, very nice stuff about language design. But we're not really talking about the language design here, aren't we?

> So yes, some documents are long, some are short, a document should be as long as it needs to be and no longer.

Correct, just remove the unrelated stuff.